### ISSUE TITLE
🐛 Cannot convert Frame with Format 34 to byte array! 

### ISSUE THREAD

**Issue by bankbiz** [cuvent/react-native-vision-camera#1781]
### What's happening?

I tried to get an image ArrayBuffer from frameProcessor with the new function introduced in v3.0.0 `frame.toArrayBuffer()`, but failed. I got the error: `Cannot convert Frame with Format 34 to byte array! `

I also try with all given pixel formats but got the same error `pixelFormat: "native" | "yuv" | "rgb"`

### Reproduceable Code

```tsx
const frameProcessor = useFrameProcessor((frame) => {
   'worklet'
   console.log("frame:", frame.width, frame.height, frame.orientation, frame.isValid) // result: 3472 3472 portrait true
   let byteImage = frame.toArrayBuffer();
}, []);

...
<Camera
   device={cameraDevice}
   isActive={true}
   frameProcessor={frameProcessor}
/>

// package.json
"dependencies": {
    "react": "18.2.0",
    "react-native": "0.72.4",
    "react-native-vision-camera": "^3.0.0",
    "react-native-worklets-core": "^0.2.0"
},
```

### Device

Xiaomi 11

### VisionCamera Version

3.0.0

### Can you reproduce this issue in the VisionCamera Example app?

- [ ] I can reproduce the issue in the VisionCamera Example app.

### Additional information

- [ ] I am using Expo
- [ ] I have enabled Frame Processors (react-native-worklets-core)
- [X] I have read the [Troubleshooting Guide](https://react-native-vision-camera.com/docs/guides/troubleshooting)
- [X] I agree to follow this project's [Code of Conduct](https://github.com/mrousavy/react-native-vision-camera/blob/main/CODE_OF_CONDUCT.md)
- [X] I searched for [similar issues in this repository](https://github.com/mrousavy/react-native-vision-camera/issues) and found none.

**Comment by barghi** [comment]
Same issue here

**Comment by jjbwii** [comment]
Same issue here. Additional info:

- When I set ‘yuv’ (format:34) my program crash
- when I set ‘rgb’ program works but GetPlanes() allways null, it seems the buffer is not available
- when I set ‘native’ it works and obvious there is no buffer available (private format)

I think there is any problem with VideoPipeline.kt Surface tranform to Image

**Comment by jlsneto** [comment]
> Same issue here. Additional info:
> 
> * When I set ‘yuv’ (format:34) my program crash
> * when I set ‘rgb’ program works but GetPlanes() allways null, it seems the buffer is not available
> * when I set ‘native’ it works and obvious there is no buffer available (private format)
> 
> I think there is any problem with VideoPipeline.kt Surface tranform to Image

Same issue.

**Comment by sarahsporck** [comment]
format 34 should be the [private](https://developer.android.com/reference/android/graphics/ImageFormat#PRIVATE) format. I guess that's also the format the emulator uses?

**Comment by Waseem6409** [comment]
@mrousavy  Can you check the issue , i am working on something but stuck due to this and also because the tflite and playtorch functionalities like imageFromFrame is not yet implemented

**Comment by mrousavy** [comment]
@Waseem6409 sure, contact us through my app development/consulting agency https://margelo.io and we can figure something out for a fair $$$.

**Comment by oliverphardman** [comment]
Same issue here with Expo.

**Comment by oliverphardman** [comment]
> Same issue here. Additional info:
> 
>     * When I set ‘yuv’ (format:34) my program crash
> 
>     * when I set ‘rgb’ program works but GetPlanes() allways null, it seems the buffer is not available
> 
>     * when I set ‘native’ it works and obvious there is no buffer available (private format)
> 
> 
> I think there is any problem with VideoPipeline.kt Surface tranform to Image

`rgb` crashes my app.

**Comment by basharmalkawi** [comment]
Same issue.

**Comment by rodgomesc** [comment]
@oliverphardman  please check if #1875 solves your crash while using `rgb`

**Comment by mrousavy** [comment]
Hey - this should be fixed in V3.2.0 as I now use the GPU HardwareBuffer APIs in YUV_420!
